Integrations/integration-DigitalShadows.yml

```diff
@@ -771,14 +780,30 @@ script:
             lastRun = {value: (new Date(nowDate.getTime() - 1*60*1000)).toISOString()};
         }
 
+        // first API call - fetch PHISHING_ATTEMPT incidents only (within the date range)
         bodyArgs = {filter_types_type:'BRAND_PROTECTION', filter_types_subTypes:'PHISHING_ATTEMPT', filter_dateRange: lastRun.value + '/' + now};
         var incidents = fetchCommandIncidents('ds-find-incidents-filtered', bodyArgs, 'PHISHING_ATTEMPT');
 
+        var ids = getIncidentsIds(incidents);
+
+        // second API call - fetch all incidents (within the date range)
         bodyArgs = {filter_dateRange: lastRun.value + '/' + now};
-        incidents = incidents.concat(fetchCommandIncidents('ds-find-incidents-filtered', bodyArgs, 'Client Incident'));
+        temp = fetchCommandIncidents('ds-find-incidents-filtered', bodyArgs, 'Client Incident');
+
+        // remove duplicated incidents from second API call that resulted in the first call
+        if (incidents) {
+            incidents = incidents.concat(temp.filter(function(inc) {
+               return ids.indexOf(JSON.parse(inc.rawJSON).id) == -1;
+            }));
+        }
+        else {
```

Comment:
Why is the 'else' case needed?

Reply:
Without separating the cases to if-else, `ids` would be a NoneType object and there will be a null pointer exception in line 796.